src/System.Security.Cryptography.Csp/tests/DSACryptoServiceProviderTests.cs

```diff
@@ -0,0 +1,294 @@
+
+                Assert.False(keyContainerInfo.HardwareDevice, "HardwareDevice");
+                Assert.False(keyContainerInfo.MachineKeyStore, "MachineKeyStore");
+                Assert.False(keyContainerInfo.Protected, "Protected");
+                Assert.False(keyContainerInfo.Removable, "Removable");
+
+                // Ephemeral keys don't successfully request the exportable bit.
+                Assert.ThrowsAny<CryptographicException>(() => keyContainerInfo.Exportable);
+
+                Assert.True(keyContainerInfo.RandomlyGenerated, "RandomlyGenerated");
+
+                Assert.Equal(KeyNumber.Signature, keyContainerInfo.KeyNumber);
+            }
+        }
+
+        [Fact]
+        public static void DefaultKey_NotPersisted()
+        {
+            using (var dsa = new DSACryptoServiceProvider())
+            {
+                Assert.False(dsa.PersistKeyInCsp);
+            }
+        }
+
+        [Fact]
+        public static void NamedKey_DefaultProvider()
+        {
+            const int KeySize = 1024;
+
+            CspParameters cspParameters = new CspParameters
+            {
+                ProviderType = PROV_DSS_DH,
+                KeyContainerName = Guid.NewGuid().ToString(),
+            };
+
+            using (new DsaKeyLifetime(cspParameters))
+            {
+                byte[] privateBlob;
+                string uniqueKeyContainerName;
+
+                using (var dsa = new DSACryptoServiceProvider(KeySize, cspParameters))
+                {
+                    Assert.True(dsa.PersistKeyInCsp, "dsa.PersistKeyInCsp");
+                    Assert.Equal(cspParameters.KeyContainerName, dsa.CspKeyContainerInfo.KeyContainerName);
+
+                    uniqueKeyContainerName = dsa.CspKeyContainerInfo.UniqueKeyContainerName;
+                    Assert.NotNull(uniqueKeyContainerName);
+                    Assert.NotEqual(string.Empty, uniqueKeyContainerName);
+
+                    privateBlob = dsa.ExportCspBlob(true);
+                    Assert.True(dsa.CspKeyContainerInfo.Exportable, "dsa.CspKeyContainerInfo.Exportable");
+                }
+
+                // Fail if the key didn't persist
+                cspParameters.Flags |= CspProviderFlags.UseExistingKey;
+
+                using (var dsa = new DSACryptoServiceProvider(cspParameters))
+                {
+                    Assert.True(dsa.PersistKeyInCsp);
+                    Assert.Equal(KeySize, dsa.KeySize);
+
+                    Assert.Equal(uniqueKeyContainerName, dsa.CspKeyContainerInfo.UniqueKeyContainerName);
+
+                    byte[] blob2 = dsa.ExportCspBlob(true);
+                    Assert.Equal(privateBlob, blob2);
+                }
+            }
+        }
+
+        [Fact]
+        public static void NamedKey_AlternateProvider()
+        {
+            const int KeySize = 512;
+
+            CspParameters cspParameters = new CspParameters(PROV_DSS_DH)
```

Comment:
That was missed clean-up on my side. Remnant from RSA which has two supported providers. Removed the test